metadata-db-app/src/cmr/metadata_db/migrations/080_add_generic_document_table.clj

```diff
@@ -0,0 +1,65 @@
+(ns cmr.metadata-db.migrations.080-add-generic-document-table
+  (:require
+   [config.mdb-migrate-helper :as h]))
+
+(defn- create-generic-documents-table
+  []
+  (h/sql
+   (clojure.string/replace
+    (str "CREATE TABLE METADATA_DB.cmr_generic_documents (
+          id NUMBER,
+          concept_id VARCHAR(255) NOT NULL,
+          native_id VARCHAR(1030) NOT NULL,
+          provider_id VARCHAR(10) NOT NULL,
+          document_name VARCHAR(20) NOT NULL,
+          schema VARCHAR(255) NOT NULL,
+          format VARCHAR(255) NOT NULL,
+          mime_type VARCHAR(255) NOT NULL,
+          metadata BLOB NOT NULL,
+          revision_id INTEGER DEFAULT 1 NOT NULL,
+          revision_date TIMESTAMP WITH TIME ZONE DEFAULT SYSTIMESTAMP NOT NULL,
+          created_at TIMESTAMP WITH TIME ZONE DEFAULT SYSTIMESTAMP NOT NULL,
+          deleted INTEGER DEFAULT 0 NOT NULL,
+          user_id VARCHAR(30),
+          transaction_id INTEGER DEFAULT 0 NOT NULL,
+
+          CONSTRAINT generic_doc_pk PRIMARY KEY (id),
+
+          CONSTRAINT gen_doc_cid_rev UNIQUE (concept_id, revision_id)
+          USING INDEX (create unique index generic_cri
+          ON cmr_generic_documents (concept_id, revision_id)))")
+    #"\s+" " ")))
+
+(defn- create-generic-document-indices
+  []
+  ;; Supports queries to find generic document revisions that are deleted
+  (h/sql
+   (clojure.string/replace
+    (str "CREATE INDEX generic_documents_crdi
+          ON METADATA_DB.cmr_generic_documents (concept_id, revision_id, deleted)")
+    #"\s+" " "))
+
+  ;; Supports queries to find generic document by native id within one provider
+  (h/sql
+   (clojure.string/replace
+    (str "CREATE INDEX generic_documents_native_id
+          ON METADATA_DB.cmr_generic_documents (native_id, provider_id)")
```

Comment:
I would use `(provider_id, native_id)` as it is likely that we also want to search by provider_id.

Reply:
There may be a need to search by provider_id, native_id and revision_id for ingest integrity check. We may need an index for that.

Reply:
are you suggesting 3 indexes? are these order dependent?

Reply:
If we always search `native_id` together with `provider_id`, one index of `(provider_id, native_id, revision_id)` would be enough. Otherwise, more index is needed.